command-parser.js

```diff
@@ -133,6 +137,233 @@ function canTalk(user, room, connection, message, targetUser) {
+		var users = this.room.users;
+		var auth = this.room.auth;
+
+		for (var i in users) {
+			var user = users[i];
+			// hardcoded for performance reasons (this is an inner loop)
+			if (user.isStaff || (auth && (auth[user.userid] || '+') !== '+')) {
+				user.sendTo(this.room, data);
+			}
+		}
+	};
+	Context.prototype.logEntry = function (data) {
+		this.room.logEntry(data);
+	};
+	Context.prototype.addModCommand = function (text, logOnlyText) {
+		this.add(text);
+		this.logModCommand(text + (logOnlyText || ""));
+	};
+	Context.prototype.logModCommand = function (result, targetRoom) {
+		if (!targetRoom) targetRoom = this.room;
+		if (!modlog[targetRoom.id]) {
+			if (targetRoom.battle) {
+				modlog[targetRoom.id] = modlog['battle'];
+			} else {
+				modlog[targetRoom.id] = fs.createWriteStream(path.resolve(__dirname, 'logs/modlog/modlog_' + targetRoom.id + '.txt'), {flags:'a+'});
+			}
+		}
+		modlog[targetRoom.id].write('[' + (new Date().toJSON()) + '] (' + targetRoom.id + ') ' + result + '\n');
+	};
+	Context.prototype.can = function (permission, target, room) {
+		if (!this.user.can(permission, target, room)) {
+			this.sendReply("/" + this.fullCmd + " - Access denied.");
+			return false;
+		}
+		return true;
+	};
+	Context.prototype.canBroadcast = function (suppressMessage) {
+		if (this.cmdToken === BROADCAST_TOKEN) {
+			var message = this.canTalk(this.originalMessage);
+			if (!message) return false;
+			if (!this.user.can('broadcast', null, this.room)) {
+				this.connection.sendTo(this.room, "You need to be voiced to broadcast this command's information.");
+				this.connection.sendTo(this.room, "To see it for yourself, use: /" + message.substr(1));
+				return false;
+			}
+
+			// broadcast cooldown
+			var normalized = message.toLowerCase().replace(/[^a-z0-9\s!,]/g, '');
+			if (this.room.lastBroadcast === normalized &&
+					this.room.lastBroadcastTime >= Date.now() - BROADCAST_COOLDOWN) {
+				this.connection.sendTo(this.room, "You can't broadcast this because it was just broadcast.");
+				return false;
+			}
+			this.add('|c|' + this.user.getIdentity(this.room.id) + '|' + (suppressMessage || message));
+			this.room.lastBroadcast = normalized;
+			this.room.lastBroadcastTime = Date.now();
+
+			this.broadcasting = true;
+		}
+		return true;
+	};
+	Context.prototype.parse = function (message, inNamespace) {
+		if (inNamespace && this.cmdToken) {
+			message = this.cmdToken + this.namespaces.concat(message.slice(1)).join(" ");
+		}
+		return CommandParser.parse(message, this.room, this.user, this.connection, this.levelsDeep + 1);
+	};
+	Context.prototype.run = function (targetCmd, inNamespace) {
+		var commandHandler;
+		if (typeof targetCmd === 'function') {
+			commandHandler = targetCmd;
+		} else if (inNamespace) {
+			commandHandler = commands;
+			for (var i = 0; i < this.namespaces.length; i++) {
+				commandHandler = commandHandler[this.namespaces[i]];
+			}
+			commandHandler = commandHandler[targetCmd];
```

Comment:
Removed ``this.commands`` as it can be easily figured out from ``this.namespaces``.